### ISSUE TITLE
[Bug]: LocalSpannerIO package namespace conflict with Beam's SpannerIO 

### ISSUE THREAD

**Issue by Abacn** [GoogleCloudPlatform/DataflowTemplates#1362]
### Related Template(s)

V1

### Template Version

N/A

### What happened?

V1 templates use LocalSpannerIO under Beam's namespace: https://github.com/GoogleCloudPlatform/DataflowTemplates/tree/main/v1/src/main/java/org/apache/beam/sdk/io/gcp/spanner and depends on beam-sdks-java-io-google-cloud-platform as well, causing unpredictable class conflict.

It happens in the 2.55.0 release validation: #1361 SpannerIO V1 templates failed with error
```
Caused by: java.lang.NoSuchMethodError: 'org.apache.beam.sdk.options.ValueProvider org.apache.beam.sdk.io.gcp.spanner.SpannerConfig.getCredentials()'
	at org.apache.beam.sdk.io.gcp.spanner.SpannerAccessor.buildSpannerOptions(SpannerAccessor.java:228)
	at org.apache.beam.sdk.io.gcp.spanner.SpannerAccessor.createAndConnect(SpannerAccessor.java:237)
	at org.apache.beam.sdk.io.gcp.spanner.SpannerAccessor.getOrCreate(SpannerAccessor.java:94)
	at org.apache.beam.sdk.io.gcp.spanner.LocalSpannerAccessor.getOrCreate(LocalSpannerAccessor.java:57)
	at com.google.cloud.teleport.templates.common.SpannerConverters$CreateTransactionFnWithTimestamp.setup(SpannerConverters.java:766
```
because SpannerAccessor was picked from beam-io-gcp but SpannerConfig was picked from v1 at run time, causing NoSuchMethodError.

We should either relocate the namespace of LocalSpannerIO or migrate to Beam's SpannerIO

**Comment by Abacn** [comment]
Even removed duplicate classes and only preserve Localxxxx srcs, there are still test failing. Found that compilation of v1/ still generates many AUtoValue_* classes: 

<img width="332" alt="image" src="https://github.com/GoogleCloudPlatform/DataflowTemplates/assets/8010435/b1ba6401-14cc-4cad-b5ad-e159ce8d06b3">

Also uploaded a reproduce at https://github.com/Abacn/codesnippets/tree/master/AutoValueTest , this might be an auto value bug

**Comment by Abacn** [comment]
As a result we cannot partially remove the duplicated sources. We either need to keep sync with Beam or remove them altogether. entered autovalue bug https://github.com/google/auto/issues/1734

**Comment by arawind** [comment]
@Abacn thanks for the ping. We're trying to get this change merged into 2.56 in https://github.com/apache/beam/pull/30893
